Makefile

```diff
@@ -685,6 +685,13 @@ chkshsgr warn-shsgr tryshsgr.c compile load
 	hasshsgr.h
 	rm -f tryshsgr.o tryshsgr
 
+hasutmpx.h: \
+tryutmpx.c compile load
+	( ( ./compile tryutmpx.c && ./load tryutmpx ) >/dev/null \
+	2>&1 \
+	&& echo \#define HASUTMPX 1 || exit 0 ) > hasutmpx.h
```

Comment:
I would still solve it as uint32.h is created currently: have 2 headers in the tree (*.h1 and *.h2) and simply copy the proper one to *.h, this solves the whole #ifdef code.

Reply:
Yes, that's better too. Have done.